### ISSUE TITLE
Further grouping of optional properties

### ISSUE THREAD

**Issue by FayCross** [thexerteproject/xerteonlinetoolkits#618]
Following on from the grouping optional properties work I did in Edinburgh, someone asked if they could also be grouped according to whether they're specific to the page or are the optional properties that can be applied to every page. Is this something we would want? Something along these lines:

![opt1](https://cloud.githubusercontent.com/assets/8926470/24962357/a87f4416-1f93-11e7-82e9-dd094b6c21cb.JPG)

Does this then look odd if there's only the generic options on a page?:

![opt2](https://cloud.githubusercontent.com/assets/8926470/24962418/d599a194-1f93-11e7-995c-5ae960e87f08.JPG)

If you would like this to be done what should the titles for the two groups be? I've used the page name for the first group but don't think 'All pages' is particularly clear for the generic ones - what would be better?

**Comment by JohnSmith-LT** [comment]
Yes I really like the look of that, even potentially building in the option to collapse certain groupings temporarily with a button

**Comment by torinfo** [comment]
Yes I would like that as well. 

Would 'General' be a good title?

**Comment by FayCross** [comment]
Ok, I'll commit it (slightly embarrassed that @torinfo's English is better than mine!)

**Comment by JohnSmith-LT** [comment]
Yes, that works for me. Great work Fay!

**Comment by ronm123** [comment]
Hi @FayCross all
Agree this looks good!
@FayCross one think I don't like about your previous work on the groupings is the 'Enable resuming' being grouped with the Tracking group. @FayCross :-(
I nearly always enable resuming and encourages others to do so too but it seems wrong that to add that you get all the SCORM options too. Perhaps others disagree with my thoughts on this and if so fine. But I'd almost prefer to have the enable resuming option mandatory but unticked by default and on it's own rather than grouped.

**Comment by JohnSmith-LT** [comment]
Yeah I agree with @ronm123 I think grouping it implies that you can only tick that if you are creating a tracked LO whereas the reasons for adding it were to allow you to bookmark and come back to a page...

Maybe the name isn't the best I could have chosen though...

**Comment by FayCross** [comment]
Ok, I'll take it out of the group

**Comment by ronm123** [comment]
Thanks @FayCross not a criticism in any way I think the new grouping etc are fantastic!

**Comment by torinfo** [comment]
lol!

I can assure you that's not true.... :-)

**Comment by FayCross** [comment]
It's not a problem Ron - I had wondered about that property (haven't used it myself) and had intended to ask everyone if it was in the right place but forgot. If anyone sees any other misplaced ones then please add or remove them from the groups - it should be pretty clear in the xwds how it's done.

**Comment by FayCross** [comment]
The properties that show in the general group are those that are taken from the basicPages.xwd file. I've added a new attribute to each of these called 'common' so that the editor can distinguish between unique & general properties.

@torinfo if you have a chance it would be great if this attribute was added to the xml by the rebuild scripts rather than having to add it to each of them manually. Not a high priority at all though as it works fine as it is as long as we remember to add the attribute to any new properties we add to basicPages.xwd
